pkgs/by-name/cl/clash-verge-rev/package.nix

```diff
@@ -1,66 +1,133 @@
-    hash = "sha256-pVEP+A4W6xLShFXuXPA6P+HZT8Hqkj/HRW2LaOOBI6U=";
+  src = fetchFromGitHub {
+    owner = "clash-verge-rev";
+    repo = "clash-verge-rev";
+    rev = "v${version}";
+    hash = "sha256-jDW5Zf6hXU5DyI4/QnzEvwMYr3PdUb8Vy69tBkxi6NI=";
   };
 
-  nativeBuildInputs = [
-    dpkg
-    wrapGAppsHook3
-    autoPatchelfHook
-  ];
-
-  buildInputs = [
-    openssl
-    webkitgtk
-  ];
+  src-service = fetchFromGitHub {
+    owner = "clash-verge-rev";
+    repo = "clash-verge-service";
+    rev = "e74e419f004275cbf35a427337d3f8c771408f07"; # no meaningful tags in this repo. The only way is updating manully every time.
+    hash = "sha256-HyRTOqPj4SnV9gktqRegxOYz9c8mQHOX+IrdZlHhYpo=";
+  };
 
-  runtimeDependencies = [
-    (lib.getLib udev)
-    libayatana-appindicator
-  ];
+  service-cargo-hash = "sha256-tjPGXsABNK3HR6QmVl97UqGkDBhqhL3sSc3vGScWmho=";
 
-  installPhase = ''
-    runHook preInstall
+  service = callPackage ./service.nix {
+    inherit
+      version
+      rustPlatform
+      src-service
+      pkg-config
+      openssl
+      webkitgtk
+      service-cargo-hash
+      pname
+      ;
+  };
 
-    mkdir -p $out/bin
-    mv usr/* $out
+  webui = callPackage ./webui.nix {
+    inherit
+      version
+      src
+      pname
+      pnpm
+      nodejs
+      stdenv
+      ;
+  };
 
-    runHook postInstall
-  '';
+  sysproxy-hash = "sha256-TEC51s/viqXUoEH9rJev8LdC2uHqefInNcarxeogePk=";
 
-  postFixup = ''
-    rm -f $out/bin/verge-mihomo
-    ln -sf ${lib.getExe clash-meta} $out/bin/verge-mihomo
-  '';
-
-  passthru.updateScript = nix-update-script { };
+  unwrapped = callPackage ./unwrapped.nix {
+    inherit
+      pname
+      version
+      src
+      sysproxy-hash
+      webui
+      meta
+      makeDesktopItem
+      pkg-config
+      rustPlatform
+      openssl
+      webkitgtk
+      lib
+      nix-update-script
+      ;
+  };
 
   meta = {
     description = "Clash GUI based on tauri";
     homepage = "https://github.com/clash-verge-rev/clash-verge-rev";
     license = lib.licenses.gpl3Plus;
     mainProgram = "clash-verge";
-    maintainers = with lib.maintainers; [ Guanran928 ];
+    maintainers = with lib.maintainers; [
+      Guanran928
+      bot-wxt1221
+    ];
     platforms = [ "x86_64-linux" ];
-    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
   };
-})
+in
+stdenv.mkDerivation {
+  inherit
+    pname
+    src
+    version
+    meta
+    ;
+  nativeBuildInputs = [
+    makeShellWrapper
+    wrapGAppsHook3
+  ];
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,share,lib/clash-verge/resources}
+
+    cp -r ${unwrapped}/share/* $out/share
+    ln -s ${service}/bin/clash-verge-service $out/bin/clash-verge-service
+    cp -r ${unwrapped}/bin/clash-verge $out/bin/clash-verge
+    ln -s ${mihomo}/bin/mihomo $out/bin/verge-mihomo
+    ln -s ${mihomo}/bin/mihomo $out/bin/verge-mihomo-alpha
```

Comment:
This is still confusing. The user sees that there is a mihomo-alpha core while it's not alpha.